### ISSUE TITLE
External diff on binary files

### ISSUE THREAD

**Issue by kayden-kim** [ForkIssues/TrackerWin#115]
Hello.

I can't compare binary files using external diff - WinMerge. WinMerge has excel plugin and I made a plugin for my custom binary file. When I compare xlsx and my binary format, I can't get correct diff in the WinMerge, It seems that file is corrupted, can't open files in temporary folder.

https://sourceforge.net/projects/winmerge/

Before v1.26, only remote files have a problem. On v1.26 both local and remote files have problems.

I suspect that files are saved by utf-8 encoding when using external diff, because I can get correct my binary file after trying to decode from UTF-8 using iconv - I have to use -c option to discard unconvertable characters. 
 
`iconv -c -f UTF-8 -t CP1250 "fobar~local.bytes" > "foobar~local2.bytes"`

I have also tried to decode xlsx file, but can't open xlsx correctly. I think iconv can't recover the file 100%, because it's binary format.

Anyway I hope that external diff is working well on binary files soon.

Thanks.

**Comment by DanPristupov** [comment]
> I think iconv can't recover the file 100%, because it's binary format.

You are right, you can not recover such binary file.

I will have a look into this.

You you fell adventurous you can try to create a custom action for files which will call your diff, it will be available in the file context menu.

You need to run a script which will save both `remote` and `local` versions and then call your merger.

```
git cat-file blob "$sha:$filepath" > local.file // checkout local version
git cat-file blob "$sha~:$filepath" > remote.file //checkout remote version
./your merger "local.file" "remote.file"
```

**Comment by kayden-kim** [comment]
@DanPristupov I'm using external diff now via custom action. Thank you so much! :)

```
@set LOCAL_PATH="%TEMP%\%~n2~local%~x2" 
@set REMOTE_PATH="%TEMP%\%~n2~remote%~x2" 
@git cat-file blob %1:%2 > %LOCAL_PATH%
@git cat-file blob %1~:%2 > %REMOTE_PATH% 
@"C:\Program Files\WinMerge\WinMergeU.exe" %LOCAL_PATH% %REMOTE_PATH% /x /wl /wr
@del %LOCAL_PATH% %REMOTE_PATH%
```

**Comment by DanPristupov** [comment]
This should be fixed in 1.27 which I just released.

**Comment by kayden-kim** [comment]
> 
> 
> This should be fixed in 1.27 which I just released.

I just checked in 1.27. Fixed. Thank you.
